File: LATIHAN4DPBO2023/Python/Sivitas.py
from Human import Human

#inisialisasi class Sivitas yang merupakan child dari class Human
class Sivitas(Human):
	# atribut
	__univ = ""
	__email = ""

	# constructor
	def __init__(self, nik = "", name = "", gender = "", univ = "", email = ""):
		super().__init__(nik, name, gender)
		self.__univ = univ
		self.__email = email

	# kumpulan method setter dan getter untuk tiap atribut
	def setUniv(self, univ):
		self.__univ = univ

	def getUniv(self):
		return self.__univ

	def setEmail(self, email):
		self.__email = email

	def getEmail(self):
		return self.__email

File: LATIHAN4DPBO2023/CPP/Sivitas.cpp
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 4
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//include file header
#include "header.hh"

//inisialisasi constructor
Sivitas::Sivitas() : Human()
{
	univ = "";
	email = "";
}

//inisialisasi constructor dengan parameter (menggunakan supper dengan class Human agar dapat memasukkan valuenya ke parameter)
Sivitas::Sivitas(string nik, string name, string gender, string univ, string email) : Human(nik, name, gender)
{
	this->univ = univ;
	this->email = email;
}

//inisialisasi method setter dan getter
void Sivitas::setUniv(string univ)
{
	this->univ = univ;
}

string Sivitas::getUniv()
{
	return this->univ;
}


void Sivitas::setEmail(string email)
{
	this->email = email;
}

string Sivitas::getEmail()
{
	return this->email;
}

//inisialisasi destructor
Sivitas::~Sivitas()
{
}